workbench_access_protect/src/Access/DeleteAccessCheck.php

```diff
@@ -0,0 +1,294 @@
+use Drupal\workbench_access\UserSectionStorage;
+use Drupal\Core\Entity\EntityFieldManager;
+use Drupal\Core\Messenger\MessengerInterface;
+use Drupal\workbench_access\Entity\AccessSchemeInterface;
+
+/**
+ * Class TaxonomyDeleteAccessCheck.
+ */
+class DeleteAccessCheck implements DeleteAccessCheckInterface {
+
+  /**
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  private $account;
+
+  /**
+   * @var \Drupal\Core\Routing\CurrentRouteMatch;
+   */
+  private $route;
+
+  /**
+   * @var \Drupal\workbench_access\UserSectionStorage
+   */
+  private $userSectionStorage;
+
+  /**
+   * @var \Drupal\Core\Entity\EntityFieldManager
+   */
+  private $fieldManager;
+
+  /**
+   * @var \Drupal\Core\Entity\EntityTypeManager
+   */
+  private $entityTypeManager;
+
+  /**
+   * @var \Drupal\Core\Entity\EntityTypeBundleInfoInterface
+   */
+  private $entityTypeBundleInfo;
+
+
+  public function __construct(CurrentRouteMatch $route,
+                              UserSectionStorage $userSectionStorage,
+                              EntityFieldManager $fieldManager,
+                              EntityTypeManager $entityTypeManager,
+                              EntityTypeBundleInfoInterface $entityTypeBundleInfo) {
+
+    $this->route = $route;
+    $this->userSectionStorage = $userSectionStorage;
+    $this->fieldManager = $fieldManager;
+    $this->entityTypeManager = $entityTypeManager;
+    $this->entityTypeBundleInfo = $entityTypeBundleInfo;
+
+  }
+
+  /**
+   * This method is used to determine if it is OK to delete.
+   *
+   * The check is based on whether or not it is being actively used for access
+   * control, and if content is assigned to it. If either of these statements
+   * is true, then 'forbidden' will be returned to prevent the term
+   * from being deleted.
+   *
+   * @return \Drupal\Core\Access\AccessResultAllowed|\Drupal\Core\Access\AccessResultForbidden
+   *   Returns 'forbidden' if the term is being used for access control.
+   *   Returns 'allowed' if the term is not being used for access control.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  public function access() {
+
+    if ($this->isDeleteAllowed()) {
+      return AccessResult::allowed();
+    }
+
+    return AccessResult::forbidden();
+  }
+
+  /**
+   * @{inheritdoc}
+   */
+  public function isDeleteAllowed(EntityInterface $entity) {
+
+    $retval = TRUE;
+
+    if ($entity instanceof EntityInterface) {
+
+      $hasAccessControlMembers = $this->doesTermHaveMembers($entity);
+      $assigned_content = $this->isAssignedToContent($entity);
+
+      /*
+       * If this entity does not have users assigned to it for access
+       * control, and the term is not assigned to any pieces of content,
+       * it is OK to delete it.
+       */
+      if ($hasAccessControlMembers || $assigned_content) {
```

Comment:
@jasonpartyka the problem I was seeing can be solved by changing this conditional so the operator is `&&` and not `||`.

I was seeing all vocabularies set not delete-able and tweaking that makes it so only vocabularies that are used for access control and have content are protected, which seems like what we want.